### DISCUSSION TITLE
how to compute jacobians derivative of the classic accelerations

### DISCUSSION THREAD

**Discussion by liuyidesktop** [stack-of-tasks/pinocchio#2192] [Q&A]
I tried the method in issue#2141 to solve the derivative of the Jacobian matrix corresponding to the classic  acceleration, but when I used the finite difference method to verify, I found that the results of the two were different. The code is as follows. Assume that the correct Jacobian derivative is dot_J, which strictly satisfies the relationship dot_J=dJ/dt. From my tests, I found that dJ_c can still solve for the correct classic acceleration, but only satisfies dJ_c@dq=dot_J@dq, however dJ_c!=dot_J, I want to know how to solve the dot_J. where dot_J=dJ/dt

```py
import numpy as np
import pinocchio as pin

model = pin.buildSampleModelManipulator()
data= model.createData()
frame_idx = model.getFrameId(model.frames[-1].name)

q=pin.neutral(model)
v=np.array([0.1,0.2,0.3,0.4,0.5,0.6])

dt=0.00001
q2=q.copy()
q2+=dt*v

pin.computeJointJacobians(model,data,q)
pin.framesForwardKinematics(model,data,q)
pin.updateFramePlacements(model, data)
pin.forwardKinematics(model, data, q,v)
pin.computeJointJacobiansTimeVariation(model,data,q,v)  


J_s=pin.getFrameJacobian(model, data, frame_idx, pin.ReferenceFrame.LOCAL_WORLD_ALIGNED)
dJ_s=pin.getFrameJacobianTimeVariation(model,data,frame_idx,pin.ReferenceFrame.LOCAL_WORLD_ALIGNED)
vs=pin.getFrameVelocity(model, data, frame_idx, pin.LOCAL_WORLD_ALIGNED)
w_cross = pin.skew(vs.angular)

J_c=J_s.copy()

#Calculate the Jacobian derivative under conventional acceleration
dJ_c = dJ_s.copy()
dJ_c[:3,:] += w_cross @ J_c[:3,:]

pin.computeJointJacobians(model,data,q2)
pin.framesForwardKinematics(model,data,q2)
pin.updateFramePlacements(model, data)
J_s2=pin.getFrameJacobian(model, data, frame_idx, pin.ReferenceFrame.LOCAL_WORLD_ALIGNED)
J_c2=J_s2.copy()

dJ_c2=(J_c2-J_c)/dt

print(dJ_c[:,4])
print(dJ_c2[:,4])
```

**Comment by ALICEYZ5** [comment]
Hi @liuxiaoyiyixiaoliu 
I got the same question. Have you been able to figure out?

**Comment by Linjackffy** [comment]
Hi @ALICEYZ5. Sorry to bother you.
Have you solved the issue regarding the incorrect results from the getFrameJacobianTimeVariation in LOCAL_WORLD_ALIGNED?

**Reply by Linjackffy** [reply]
With the latest devel branch, this issue has been resolved.See[https://github.com/stack-of-tasks/pinocchio/discussions/2463#discussion-7363158](https://github.com/stack-of-tasks/pinocchio/discussions/2463#discussion-7363158)
